### ISSUE TITLE
Fatal error binding nvidia persistenced socket

### ISSUE THREAD

**Issue by dtrudg** [apptainer/singularity#4702]
### Version of Singularity:

What version of Singularity are you using? Run:

```
3.5.0-rc.1+4-ga7c731b22
```

### Expected behavior

On a machine with the nvidia-persistenced running, a container starts with `--nv` correctly.

This works correctly on 3.4.2:

```
dave@ythel~> singularity run --nv docker://ubuntu
dave@ythel:~
10:27 AM $ 
```

### Actual behavior

On a machine with the nvidia-persistenced running, a container does not start with `--nv` correctly

```
dave@ythel~> singularity run --nv docker://ubuntu
...
INFO:    Creating SIF file...
FATAL:   container creation failed: mount /var/run/nvidia-persistenced/socket->/var/run/nvidia-persistenced/socket error: while mounting /var/run/nvidia-persistenced/socket: destination /var/run/nvidia-persistenced/socket doesn't exist in container
```

### Steps to reproduce this behavior

Try using `--nv` on a machine with nvidia drivers installed *and* the `nvidia-persistenced` running

### What OS/distro are you running

```
Debian 10.1
```

### How did you install Singularity

Git tree

**Comment by dtrudg** [comment]
@cclerget reports it is working on Ubuntu.

On my non-working Debian 10.1 the socket is in `/run` and `/var/run` is a symlink.

Looking at 3.4.2 - it has the same mount error, but it's not fatal - only shows on debug:

```
DEBUG   [U=1000,P=18777]   mountGeneric()                Mounting /var/run/nvidia-persistenced/socket to /usr/local/var/singularity/mnt/session/final/run/nvidia-persistenced/socket
DEBUG   [U=1000,P=18777]   mountGeneric()                Skipping mount, /usr/local/var/singularity/mnt/session/final/run/nvidia-persistenced/socket doesn't exist in container
```

**Comment by dtrudg** [comment]
Noting that for Cedric it worked as his system had the nvidia container cli installed, while mine was using the nvliblist.

This is now fixed via #4704

```
dave@ythel~/S/G/s/builddir> singularity version
3.5.0-rc.1+14-g77612731d
dave@ythel~/S/G/s/builddir> singularity run --nv docker://ubuntu
Singularity> ls /var/run/nvidia-persistenced/socket
/var/run/nvidia-persistenced/socket
Singularity> ls -lah /var
total 5.0K
drwxr-xr-x 11 root root  160 Oct 29 16:25 .
drwxr-xr-x  1 dave dave  120 Nov  7 09:19 ..
drwxr-xr-x  2 root root    3 Apr 24  2018 backups
drwxr-xr-x  5 root root   57 Oct 29 16:25 cache
drwxr-xr-x  7 root root   88 Apr  5  2018 lib
drwxrwsr-x  2 root root    3 Apr 24  2018 local
lrwxrwxrwx  1 root root    9 Oct 29 16:25 lock -> /run/lock
drwxr-xr-x  3 root root  157 Oct 29 16:25 log
drwxrwsr-x  2 root root    3 Oct 29 16:25 mail
drwxr-xr-x  2 root root    3 Oct 29 16:25 opt
lrwxrwxrwx  1 root root    4 Oct 29 16:25 run -> /run
drwxr-xr-x  2 root root   27 Oct 29 16:25 spool
drwxrwxrwt  3 root root 4.0K Nov  7 09:16 tmp
```
